src/ucp/wireup/address.c

```diff
@@ -216,17 +216,18 @@ ucp_address_gather_devices(ucp_worker_h worker, ucp_ep_h ep, uint64_t tl_bitmap,
 
         dev = ucp_address_get_device(context, rsc_index, devices, &num_devices);
 
-        if ((flags & UCP_ADDRESS_PACK_FLAG_EP_ADDR) &&
-            ucp_worker_iface_is_tl_p2p(iface_attr)) {
-            /* Each lane which matches the resource index adds an ep address
-             * entry. The length and flags is packed in non-unified mode only.
-             */
+        if (flags & UCP_ADDRESS_PACK_FLAG_EP_ADDR) {
```

Comment:
why did you break down this 'if'?

Reply:
assertion below suppresses false positive of static analizers for ep dereference

Reply:
but this assertion was here before this change. why didn't it fail before?

Reply:
because the ep was not accessed is 2nd part of condition before.
ucp_worker_iface_is_tl_p2p(iface_attr, ucp_ep_has_cm_lane(**ep**))